crates/dyn-abi/src/eip712/resolver.rs

```diff
@@ -316,7 +316,8 @@ impl Resolver {
 
     /// Ingest a sol struct typedef.
     pub fn ingest_sol_struct<S: SolStruct>(&mut self) {
-        self.ingest_string(S::eip712_encode_type()).unwrap();
+        self.ingest_string(<S as SolStruct>::eip712_encode_type())
```

Comment:
Both `SolType` and `SolStruct` have a `fn eip712_encode_type`

Reply:
while we're making a large breaking change to the encode interface, lets also fix this by differentiating the names

Reply:
what should the names be?

Reply:
Do we need them on both interfaces, or can we have `eip712_encode_type` on only `SolType` @DaniPopes

Reply:
`SolType`'s returns `Some` only for structs, so it should only be on `SolStruct`

Reply:
Removed in 820ee0750ada1ed4b35ed6bea7236d71e7aeb653